tests/shell/test_posix.py

```diff
@@ -192,33 +182,38 @@ def test_basic_integration(
         )
         sh.expect(r"Executing transaction: ...working... done.*\n")
         sh.assert_env_var("?", "0", use_exact=True)
+        sh.sendline('cat /proc/$$/status | grep "^Pid:"')
 
+        # see tests/test-recipes/small-executable
         sh.sendline("small")
         sh.expect_exact("Hello!")
 
         # TODO: assert that reactivate worked correctly
 
         sh.sendline("type conda")
-        sh.expect(conda_is_a_function)
+        sh.expect(is_a_function(sh))
 
+        # see tests/test-recipes/small-executable
         sh.sendline(f"conda run {dev_arg} small")
         sh.expect_exact("Hello!")
 
         # regression test for #6840
-        sh.sendline(f"conda {install} --blah")
-        sh.assert_env_var("?", "2", use_exact=True)
-        sh.sendline("conda list --blah")
-        sh.assert_env_var("?", "2", use_exact=True)
+        sh.sendline(f'conda {install} --blah || echo "exitcode=$?"')
+        sh.expect_exact("error: unrecognized arguments: --blah")
+        sh.expect_exact("exitcode=2")
+        sh.sendline('conda list --blah || echo "exitcode=$?"')
+        sh.expect_exact("error: unrecognized arguments: --blah")
+        sh.expect_exact("exitcode=2")
```

Comment:
Hm so something with `bash` running in the GH runner on Windows causes `echo $?` to not work as a standalone call after running `conda install --blah`, no idea why but doing this instead works...

Reply:
I would assume GitHub has a post-processor instead of trying to override `echo` `printf` etc.

Reply:
weirdly an earlier `echo $?` in the test works so 🤷🏼‍♂️